_source/_posts/2022-04-21-jhipster-k8s-azure.md

```diff
@@ -0,0 +1,667 @@
+
+```
+brew install kubeseal
+```
+
+Encrypt `ENCRYPT_KEY` and `SPRING_DATA_MONGODB_URI` (the Cosmos DB connect string). Run the following command to do this, **replacing the two values with your encryption key and your connection string**.
+
+```
+kubectl create secret generic project-secrets \
+  --from-literal=ENCRYPT_KEY='<your-encryption-key>' \
+  --from-literal=SPRING_DATA_MONGODB_URI='<your-connection-string>' \
+  --dry-run=client -o yaml > secrets.yml
+```
+
+If you deleted the namespace earlier, you need to create it again.
+
+```bash
+kubectl apply -f namespace.yml
+```
+
+Now, use `kubeseal` to convert the secrets to encrypted secrets.
+
+```
+kubeseal --cert tls.crt --format=yaml -n demo < secrets.yml > sealed-secrets.yml
+```
+
+Remove the original secrets file and deploy your sealed secrets.
+
+```
+rm secrets.yml
+kubectl apply -n demo -f sealed-secrets.yml && kubectl get -n demo sealedsecret project-secrets
+```
+
+You need to update the `yml` files to refer to the encrypted values. Add the following env variable to the `jhister-registry` container in `k8s/registry-k8s/jhipster-registry.yml` (if an `ENCRYPT_KEY` already exists, replace it).
+
+```yml
+env:
+...
+  - name: ENCRYPT_KEY
+    valueFrom:
+      secretKeyRef:
+        name: project-secrets
+        key: ENCRYPT_KEY
+```
+
+In `k8s/store-k8s/store-deployment.yml`, change the `SPRING_DATA_MONGODB_URI` env variable to use the sealed secret.
+
+```yaml
+env:
+...
+- name: SPRING_DATA_MONGODB_URI
+  valueFrom:
+    secretKeyRef:
+      name: project-secrets
+      key: SPRING_DATA_MONGODB_URI
+```
+
+Deploy the cluster.
+
+```bash
+./kubectl-apply.sh -f
+```
+
+Give the cluster a bit to start. Check it with one of the tools I mentioned or the following:
+
+```bash
+kubectl get pods -n demo
+```
+
+Once everything is ready, port forward the registry to check the services.
+
+```bash
+kubectl port-forward svc/jhipster-registry -n demo 8761
+```
+
+Make sure you can log in and that all the services are green. You should either be automatically logged in and redirected back to the home page or directed to log in with the Okta login screen.
+
+This is the happy dance.
+
+{% img blog/jhipster-k8s-azure/happy-dance.png alt:"Happy Dance!" width:"600" %}{: .center-image }
+
+
+Forward the gateway and test the app.
+
+```bash
+kubectl port-forward svc/gateway -n demo 8080
+```
+
+Log in. Make sure everything works. Kick the tires. Start a blog. Create some products. Influence some people. Restore democracy. Take over the world. Whatever you want.
+
+Once you're done with everything, you can delete the resource group. This will also delete all of the resources in the resource group, including the Cosmos database and the AKS cluster.
+
+```bash
+az aks delete --name jhipster-demo --resource-group australia-east --no-wait --yes
+```
+
+#### Wrapping Up
```

Comment:
Don't miss the opportunity for keywords in a header! This should be an h2 (##) as well.